packages/storage/src/providers/s3/apis/getUrl.ts

```diff
@@ -61,26 +62,34 @@ export const getUrl = async function (
 	const getUrlOptions = {
 		accessLevel,
 		credentials,
-		expiration: options?.expiration ?? DEFAULT_PRESIGN_EXPIRATION,
+		expiration: options?.expiresIn ?? DEFAULT_PRESIGN_EXPIRATION,
 		signingRegion: region,
 		region,
 		signingService: S3_SERVICE_NAME,
 	};
 
-	let urlExpiration = options?.expiration ?? DEFAULT_PRESIGN_EXPIRATION;
+	let urlExpirationInMS = options?.expiresIn ?? DEFAULT_PRESIGN_EXPIRATION;
+	const awsCredExpiration = credentials?.expiration;
+	if (awsCredExpiration)
+		if (awsCredExpiration) {
+			const awsCredExpirationInSec = Math.floor(
+				(awsCredExpiration?.getTime() - Date.now()) / 1000
+			);
+			urlExpirationInMS =
+				awsCredExpirationInSec < urlExpirationInMS
+					? awsCredExpirationInSec
+					: urlExpirationInMS;
```

Comment:
`urlExpirationInMS` should be `urlExpirationInSec`.
```suggestion
        Math.min(awsCredExpirationInSec, urlExpirationInSec)
```

Reply:
This is cool. Thanks for the suggestion